### PR TITLE
ci(tests): Commit snapshot changes on failure

### PR DESCRIPTION
**Pull request by underyx**
PR checklist:
- [x] Documentation is up-to-date
- [x] Changelog is up-to-date
- [x] Change has no security implications (otherwise, ping security team)

### CODE REVIEW

**Comment by underyx** [comment]
The last snapshot commit (https://github.com/returntocorp/semgrep/commit/c1183b454e53026deff5ba27f3c5bf9b2aaf1693) included some additional build artifact files.

So I changed the job to limit which directory is [passed to `git add` like documented](https://github.com/EndBug/add-and-commit#inputs): `+ add: semgrep/tests/e2e/snapshots`. I'm not gonna set up another round of testing this with a fake snapshot failure and than changing the PR back cause it'd be another 40 minutes :weary:

File: .github/workflows/tests.yml [hunk 1]
```diff
@@ -137,14 +140,43 @@ jobs:
...
+          add: semgrep/tests/e2e/snapshots
+          default_author: github_actions
+          message: "Update pytest snapshots"
+          new_branch: snapshot-updates-${{ github.run_id }}-${{ github.run_attempt }}
+      - name: Comment about any snapshot updates
+        if: ${{ failure() && steps.snapshot-commit.outputs.pushed == 'true' }}
+        run: |
+          echo ":camera_flash: The pytest shapshots changed in your PR" >> /tmp/message.txt
+          echo >> /tmp/message.txt
+          echo "1. Review the changes at https://github.com/returntocorp/semgrep/commit/${{ steps.snapshot-commit.outputs.commit_long_sha }}" >> /tmp/message.txt
```
Comment (nbrahms) [inline .github/workflows/tests.yml:173]:
```suggestion
          echo "Please carefully review these changes and make sure they are intended:" >> /tmp/message.txt
          echo "1. Review the changes at https://github.com/returntocorp/semgrep/commit/${{ steps.snapshot-commit.outputs.commit_long_sha }}" >> /tmp/message.txt
```

**Review by emjin — APPROVED** [review]
Looks good to me! Are there any security implications associated with giving write access?

**Review by brendongo — APPROVED** [review]
Like how it automates generating but requires manual work to get changes in the PR

File: .github/workflows/tests.yml [hunk 2]
```diff
@@ -137,14 +140,43 @@ jobs:
...
+        with:
+          add: semgrep/tests/e2e/snapshots
+          default_author: github_actions
+          message: "Update pytest snapshots"
+          new_branch: snapshot-updates-${{ github.run_id }}-${{ github.run_attempt }}
+      - name: Comment about any snapshot updates
+        if: ${{ failure() && steps.snapshot-commit.outputs.pushed == 'true' }}
+        run: |
+          echo ":camera_flash: The pytest shapshots changed in your PR" >> /tmp/message.txt
+          echo >> /tmp/message.txt
```
Comment (underyx) [inline .github/workflows/tests.yml]:
```suggestion
          echo ":camera_flash: The pytest shapshots changed in your PR." >> /tmp/message.txt
          echo "Please carefully review these changes and make sure they are intended:" >> /tmp/message.txt
          echo >> /tmp/message.txt
```

File: .github/workflows/tests.yml (hunk 1 above)
Comment (underyx) [reply .github/workflows/tests.yml:173]: applied your suggestion with slightly different spacing!

**Comment by underyx** [comment]
> Looks good to me! Are there any security implications associated with giving write access?

Thanks for calling this out @emjin! Actually by default [the token has write access to _everything_](https://docs.github.com/en/actions/security-guides/automatic-token-authentication#permissions-for-the-github_token). So by listing the two write permissions here, we're making it more restrictive.

**Pull request merged**